sensor/kubernetes/fake/vm_index_reports.go

```diff
@@ -0,0 +1,330 @@
+)
+
+const (
+	precomputedReportVariants = 5
+	// reportIntervalJitterPercent defines the percentage of random jitter to apply to report intervals.
+	// With 0.05 (5%), a 60s interval will vary between 57s-63s, making timing more realistic.
+	reportIntervalJitterPercent = 0.05
+
+	// listenerRestartDelay is the time to wait for the listener restart to complete before populating VMs.
+	// The 2s delay is chosen to be longer than typical listener restart time (~100-200ms) to provide a safety margin.
+	// This is acceptable for fake workload code (but not production code).
+	listenerRestartDelay = 2 * time.Second
+)
+
+type vmInfo struct {
+	id       string
+	vsockCID uint32
+	name     string
+}
+
+type reportTemplate struct {
+	packages     map[string]*v4.Package
+	repositories map[string]*v4.Repository
+}
+
+// reportGenerator generates fake VM index reports using pre-built templates
+type reportGenerator struct {
+	templates          []reportTemplate
+	currentTemplateIdx uint32
+}
+
+func newReportGenerator(numPackages, numRepos int) *reportGenerator {
+	variantCount := precomputedReportVariants
+	if variantCount <= 0 {
+		variantCount = 1
+	}
+
+	templates := make([]reportTemplate, variantCount)
+	for variant := range variantCount {
+		repositories := make(map[string]*v4.Repository, numRepos)
+		for i := range numRepos {
+			repoID := fmt.Sprintf("repo-template-%d-%d", variant, i)
+			repositories[repoID] = &v4.Repository{
+				Id:   repoID,
+				Name: fmt.Sprintf("repository-%d", i),
+				Uri:  fmt.Sprintf("https://repo%d.example.com", i),
+				Key:  fmt.Sprintf("key-%d", i),
+				Cpe:  "cpe:2.3:o:redhat:enterprise_linux:9:*:fastdatapath:*:*:*:*:*", // valid CPE to also load scanner V4 matcher.
+			}
+		}
+
+		packages := make(map[string]*v4.Package, numPackages)
+		for i := range numPackages {
+			pkgID := fmt.Sprintf("pkg-template-%d-%d", variant, i)
+
+			packages[pkgID] = &v4.Package{
+				Id:             pkgID,
+				Name:           fmt.Sprintf("package%d", i),
+				Version:        fmt.Sprintf("1.%d.%d", i/10, i%10),
+				Kind:           "binary",
+				Arch:           "amd64",
+				RepositoryHint: "hash:sha256:f52ca767328e6919ec11a1da654e92743587bd3c008f0731f8c4de3af19c1830|key:199e2f91fd431d51",
+				Cpe:            "cpe:2.3:o:redhat:enterprise_linux:9:*:fastdatapath:*:*:*:*:*", // valid CPE to also load scanner V4 matcher.
+				PackageDb:      "sqlite:usr/share/rpm",
+				Source: &v4.Package{
+					Id:      pkgID,
+					Name:    fmt.Sprintf("package%d", i),
+					Version: fmt.Sprintf("1.%d.%d", i/10, i%10),
+					Kind:    "source",
+				},
+				NormalizedVersion: &v4.NormalizedVersion{
+					Kind: "rpm",
+					V:    []int32{1, 0, 0},
+				},
+				Module: fmt.Sprintf("module%d", i),
+			}
+		}
+
+		templates[variant] = reportTemplate{
+			packages:     packages,
+			repositories: repositories,
+		}
+	}
+
+	return &reportGenerator{
+		templates: templates,
+	}
+}
+
+func (g *reportGenerator) nextTemplate() reportTemplate {
```

Comment:
Consider using a mutex so that there are not multiple threads updating `currentTemplateIdx` at the same time.